src/browser/commands.js

```diff
@@ -0,0 +1,39 @@
+import { sendKeys, sendMouse } from '@web/test-runner-commands';
+
+function getElementPosition(elem) {
+	const { x, y, width, height } = elem.getBoundingClientRect();
+	return [
+		Math.floor(x + window.scrollX + width / 2),
+		Math.floor(y + window.scrollY + height / 2),
+	];
+}
+
+export async function click(elem) {
+	await sendMouse({ type: 'click', position: getElementPosition(elem) });
+}
+
+export async function clickAt(x, y) {
+	await sendMouse({ type: 'click', position: [x, y] });
+}
+
+export async function focus(elem) {
+	await sendKeys({ press: 'Shift' }); // Tab moves focus, Escape causes dismissible things to close
+	elem.focus({ focusVisible: true });
+}
+
+export async function hover(elem) {
+	await sendMouse({ type: 'move', position: getElementPosition(elem) });
+}
+
+export async function hoverAt(x, y) {
+	await sendMouse({ type: 'move', position: [x, y] });
+}
+
+export async function keyboard(action, keys, elem) {
+	if (elem) {
+		await focus(elem);
+	}
+	const val = {};
+	val[action] = keys;
```

Comment:
I found the existing API a bit confusing, where you had to call it like `sendKeys({ press: 'A' })` or `sendKeys({ type: '123' })`. This splits out the action (type/press/down/up) from the key as separate parameters.

Open to keeping it the other way around though, it just seemed weird that you could go `sendKeys({ press: 'A', type: 'b' })` and it would explode.

Reply:
Yikes... trying to use these in a test and I think most of the names are reserved since they exist already on `window` -- i.e. `window.focus()`, `window.hover()`, etc. Will try something else...